collector/receiver/telemetryapireceiver/receiver.go

```diff
@@ -161,26 +184,118 @@ func (r *telemetryAPIReceiver) httpHandler(w http.ResponseWriter, req *http.Requ
+	scopeLog.Scope().SetName(scopeName)
+	for _, el := range slice {
+		r.logger.Debug(fmt.Sprintf("Event: %s", el.Type), zap.Any("event", el))
+		logRecord := scopeLog.LogRecords().AppendEmpty()
+		logRecord.Attributes().PutStr("type", el.Type)
+		if t, err := time.Parse(timeFormatLayout, el.Time); err == nil {
+			logRecord.SetTimestamp(pcommon.NewTimestampFromTime(t))
+			logRecord.SetObservedTimestamp(pcommon.NewTimestampFromTime(time.Now()))
+		} else {
+			r.logger.Error("error parsing time", zap.Error(err))
+			return plog.Logs{}, err
+		}
+		if el.Type == string(telemetryapi.Function) || el.Type == string(telemetryapi.Extension) {
+			if record, ok := el.Record.(map[string]interface{}); ok {
+				// in JSON format https://docs.aws.amazon.com/lambda/latest/dg/telemetry-schema-reference.html#telemetry-api-function
+				if timestamp, ok := record["timestamp"].(string); ok {
+					if t, err := time.Parse(timeFormatLayout, timestamp); err == nil {
+						logRecord.SetTimestamp(pcommon.NewTimestampFromTime(t))
+					} else {
+						r.logger.Error("error parsing time", zap.Error(err))
+						return plog.Logs{}, err
+					}
+				}
+				if level, ok := record["level"].(string); ok {
+					logRecord.SetSeverityNumber(severityTextToNumber(level))
+					logRecord.SetSeverityText(logRecord.SeverityNumber().String())
+				}
+				if requestId, ok := record["requestId"].(string); ok {
+					logRecord.Attributes().PutStr(semconv.AttributeFaaSInvocationID, requestId)
+				}
+				if line, ok := record["message"].(string); ok {
+					logRecord.Body().SetStr(line)
+				}
+			} else {
+				// in plain text https://docs.aws.amazon.com/lambda/latest/dg/telemetry-schema-reference.html#telemetry-api-function
+				if line, ok := el.Record.(string); ok {
+					logRecord.Body().SetStr(line)
+				}
+			}
+		}
+	}
+	return log, nil
+}
+
+func severityTextToNumber(severityText string) plog.SeverityNumber {
+	mapping := map[string]plog.SeverityNumber{
+		"TRACE":    plog.SeverityNumberTrace,
+		"TRACE2":   plog.SeverityNumberTrace2,
+		"TRACE3":   plog.SeverityNumberTrace3,
+		"TRACE4":   plog.SeverityNumberTrace4,
+		"DEBUG":    plog.SeverityNumberDebug,
+		"DEBUG2":   plog.SeverityNumberDebug2,
+		"DEBUG3":   plog.SeverityNumberDebug3,
+		"DEBUG4":   plog.SeverityNumberDebug4,
+		"INFO":     plog.SeverityNumberInfo,
+		"INFO2":    plog.SeverityNumberInfo2,
+		"INFO3":    plog.SeverityNumberInfo3,
+		"INFO4":    plog.SeverityNumberInfo4,
+		"WARN":     plog.SeverityNumberWarn,
+		"WARN2":    plog.SeverityNumberWarn2,
+		"WARN3":    plog.SeverityNumberWarn3,
+		"WARN4":    plog.SeverityNumberWarn4,
+		"ERROR":    plog.SeverityNumberError,
+		"ERROR2":   plog.SeverityNumberError2,
+		"ERROR3":   plog.SeverityNumberError3,
+		"ERROR4":   plog.SeverityNumberError4,
+		"FATAL":    plog.SeverityNumberFatal,
+		"FATAL2":   plog.SeverityNumberFatal2,
+		"FATAL3":   plog.SeverityNumberFatal3,
+		"FATAL4":   plog.SeverityNumberFatal4,
+		"CRITICAL": plog.SeverityNumberFatal,
```

Comment:
Adding `CRITICAL` here. It is because Python `logging` library can log `CRITICAL` (50) and it is mapped to Fatal in opentelmetry-python https://opentelemetry-python.readthedocs.io/en/latest/_modules/opentelemetry/_logs/severity.html#std_to_otel